web/pdf_find_controller.js

```diff
@@ -1052,6 +1056,15 @@ class PDFFindController {
   }
 
   #updateUIState(state, previous = false) {
+    if (
+      !this.#updateMatchesCountOnProgress &&
+      (!this.#hasTotalCount || state === FindState.PENDING)
```

Comment:
Rather than adding even more internal state, that we need to keep in sync, could we perhaps just utilize the existing field instead?
```suggestion
      (this.#visitedPagesCount !== this._linkService.pagesCount ||
        state === FindState.PENDING)
```

Reply:
I tested locally in setting `updateMatchesCountOnProgress` to `false` in the ctor and with some debug two `update` functions and I don't see anything wrong but I can overlook something.
And you're right but we send a `updatefindmatchescount` when everything is done then from the `total` that something has been found or not so it shouldn't matter to not have the event.

Reply:
I don't know how the GeckoView side of things work, so I simply want to highlight something that I noticed while looking at the code.

Reply:
Basically the find UI send an event for any change with a callback and this one can be used only one time to send back a response hence the goal of this patch (and the other one before this one) is to avoid to have any temporary result.
And yesterday I was trying to write a test for GV and the find feature, I struggled a bit to finally discover that at the beginning we send a `FOUND, { count: 0, total: 0 }` which is probably something wrong (I mean it doesn't make sense) but in a context of a live update it isn't awful.
After the tests I'm writing for GV (and maybe other bug fixes) we should have something good for Fenix.